Cargo.toml

```diff
@@ -48,6 +48,10 @@ prost = "0.12.1"
 log = "0.4.20"
 itertools = "0.12.1"
 serial_test = "3.0.0"
+aes-gcm = "0.10.3"
```

Comment:
Very good point and thank you for the validation, I was thinking about this just today when testing this draft e2e. I'll move forward with using `openssl` instead (tbh I was't 100% sure about those libs either, I would have had to do some more digging to ensure they were well-maintained etc etc) :+1:

Reply:
I've [replaced](https://github.com/rh-ecosystem-edge/recert/pull/160/files#diff-8c4912bab47ef7cc1eddfef6a5ce4017becc2dd2f69bf098e09f8d48dbfcdc19R31-R185) Rust libs with `openssl` for the AES-CBC mode and it doesn't seem to be that slower (or maybe even has the same performance, I'll get some numbers to support this). 

But we can't do the same for AES-GCM, as per [this](https://github.com/openssl/openssl/blob/260ecea0d4e46d63464636405f9925ef65d0747e/doc/man1/openssl-enc.pod.in#L299-L314) openssl doc (I tested it just to be sure that this is actually the case). TL:DR openssl doesn't and won't support AEAD ciphers, because it can't validate the authentication tag until the very end of the stream, at which point it can't rollback all the data that it has already deciphered. 

I'm now looking into other FIPS compliant options. Please let me know if you have any ideas.

Reply:
When looking into FIPS, OpenShift etcd encryption and openssl, I'm assuming that: 

since Go libs are not FIPS compliant and are replaced by openssl and openssl doesn't support AES-GCM, the OCP etcd AES-GCM encryption is by default NOT FIPS compliant. 

So we can use the AES Rust libs to support AES-GCM etcd encryption, but we should document that using AES-GCM is NOT FIPS compliant. WDYT?

Reply:
Resolved offline that this makes sense :+1: